### PR TITLE
✅ test(dok-86): add initial cypress testing

### PR DESCRIPTION
**Pull request by larwaa**
## Proposed Changes

- Add Cypress tests for E2E-testing of the application

# To do
- [x] Log in to Feide using requests
- [ ] Add Babel plugin for removing data-test-ids for production builds
- [ ] Merge with #320 

## Types of Changes

- [ ] New feature
- [ ] Bug fix
- [ ] Enhancement/optimization
- [ ] Refactor
- [ ] Style
- [x] Build/dependencies
- [ ] Other

## Issue(s) fixed or closed by this PR

- DOK-86

## Checklist

- [ ] I have added tests to cover my changes (for features/bugs)
- [ ] I am pleased with the readability of the code (if not, state where you need input)
- [ ] I have tested the changes and verified that they work and do not break anything (to the best of my ability)

### CODE REVIEW

File: docker-compose.integration.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      - SENTRY_DSN=$SENTRY_DSN
+      - SENTRY_ENVIRONMENT=$SENTRY_ENVIRONMENT
+      - SENTRY_TRACES_SAMPLE_RATE=$SENTRY_TRACES_SAMPLE_RATE
+      - SENTRY_SEND_DEFAULT_PII=$SENTRY_SEND_DEFAULT_PII
+      - SECRET_KEY=$DJANGO_SECRET_KEY
+      - DB_NAME=$DB_NAME
+      - DB_USER=$DB_USER
+      - DB_PASSWORD=$DB_PASSWORD
+      - DB_HOST=$DB_HOST
+      - DB_PORT=$DB_PORT
```
Comment (larwaa) [inline docker-compose.integration.yml:51]: This is highly inconvenient, possibly pass an environment file instead?

File: docker-compose.integration.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  frontend:
+    build:
+      context: ./frontend
+      dockerfile: Dockerfile.prod
+      args:
+        - NEXT_PUBLIC_GRAPHQL_BACKEND_URI=$NEXT_PUBLIC_GRAPHQL_BACKEND_URI
+        - NEXT_PUBLIC_DATAPORTEN_ID=$NEXT_PUBLIC_DATAPORTEN_ID
+        - NEXT_PUBLIC_DATAPORTEN_REDIRECT_URI=$NEXT_PUBLIC_DATAPORTEN_REDIRECT_URI
+        - NEXT_PUBLIC_DATAPORTEN_STATE=$NEXT_PUBLIC_DATAPORTEN_STATE
+        - NEXT_PUBLIC_FRONTEND_URI=$NEXT_PUBLIC_FRONTEND_URI
```
Comment (larwaa) [inline docker-compose.integration.yml:64]: Inconvenient, and can be removed be re-utilizing a previously built image. See DOK-248.

File: frontend/package.json [hunk 3]
```diff
@@ -68,7 +68,7 @@
     "typescript": "^4.3.4"
   },
   "lint-staged": {
-    "src/**/*.{ts,tsx}": [
+    "../**/*/.{ts,tsx}": [
```
Comment (larwaa) [inline frontend/package.json:71]: Ensure also E2E ts-files are formatted properly

File: backend/config/settings/production.py [hunk 4]
```diff
@@ -71,8 +71,8 @@
 GRAPHENE["MIDDLEWARE"] += ["config.sentry.middleware.SentryMiddleware"]  # noqa
 
 # Sentry
-SENTRY_DSN = cast(str, env("SENTRY_DSN"))
-SENTRY_LOG_LEVEL = cast(int, env.int("DJANGO_SENTRY_LOG_LEVEL", logging.INFO))
+SENTRY_DSN: str = env("SENTRY_DSN")
+SENTRY_LOG_LEVEL: int = cast(int, env.int("DJANGO_SENTRY_LOG_LEVEL", logging.INFO))
```
Comment (hermannm) [inline backend/config/settings/production.py:75]:
```suggestion
SENTRY_LOG_LEVEL: int = env.int("DJANGO_SENTRY_LOG_LEVEL", logging.INFO)
```
Unnecessary cast?

**Review by hermannm — CHANGES_REQUESTED** [review]

File: backend/config/settings/production.py (hunk 4 above)
Comment (larwaa) [reply backend/config/settings/production.py:75]: No, the Django Environ package is poorly type annotated, and does not return the proper type. Without casting it, the IDE will yell at us for claiming that it's an `int`.

**Pull request closed without merging**